### DISCUSSION TITLE
Curl works in postman but not in action

### DISCUSSION THREAD

**Discussion by gregjwoffindin** [community/community#23627] [Projects and Issues]
<p>Using curl in Postman I can get the results of my repo runs:<br>
curl --location --request GET ‘<a href="https://api.github.com/repos/%7B%7Borg%7D" rel="noopener nofollow ugc">https://api.github.com/repos/{{org}</a>}/{{my_repo}}/actions/runs’ <br>
–header ‘Accept: application/vnd.github.everest-preview+json’ <br>
–header ‘Content-Type: application/json’ <br>
–header 'Authorization: token XXXXXXXX <br>
–header ‘cache-control: no-cache’<br>
however the same curl in my github action produces errors</p>
<h1>
<a name="curl-results-for-last-automation-run-1" class="anchor" href="#curl-results-for-last-automation-run-1"></a>Curl results for last automation run</h1>
<pre><code>  - name: curl for to pull automation results
    run:      curl -X GET 'https://api.github.com/repos/{{org}}/{{my_repo}}/actions/runs' \
     -H "Accept:application/vnd.github.v3+json" \
     -H "Content-Type:application/json" \
     -H "Authorization:token ${{ secrets.XXXXXXXX }} "   
     STATUS=$(jq -r ".workflow_runs | sort_by( .created_at ) | .[-1] | .status"  ) 
     echo "Check suite state:${STATUS}"
</code></pre>
<p>Error:<br>
% Total % Received % Xferd Average Speed Time Time Time Current<br>
Dload Upload Total Spent Left Speed<br>
0 0 0 0 0 0 0 0 --:–:-- --:–:-- --:–:-- 0<br>
100 138 100 138 0 0 2262 0 --:–:-- --:–:-- --:–:-- 2262<br>
curl: (3) URL using bad/illegal format or missing URL<br>
curl: (3) URL using bad/illegal format or missing URL<br>
curl: (3) URL using bad/illegal format or missing URL<br>
{<br>
“message”: “Not Found”,<br>
“documentation_url”: “<a href="https://docs.github.com/rest/reference/actions#list-workflow-runs-for-a-repository" rel="noopener nofollow ugc">https://docs.github.com/rest/reference/actions#list-workflow-runs-for-a-repository</a>”<br>
}</p>
<p>I have already tried regenerating PAT, using token vs bearer .   Any suggestions?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The <code>run</code> part is a plain YAML string, not a block, so the newlines probably don’t work the way you assume they do. I suspect that results in curl trying to interpret the backslashes.</p>
<p>Other than that I’d tidy up the command line to reduce the risk of subtle errors, in particular the <code>-X GET</code> is pointless because it’s the default method anyway, and clean up the header format. So, try:</p>
<pre><code class="lang-yaml">  - name: curl for to pull automation results
    run: |
      curl 'https://api.github.com/repos/{{org}}/{{my_repo}}/actions/runs' \
        -H "Accept: application/vnd.github.v3+json" \
        -H "Content-Type: application/json" \
        -H "Authorization: token ${{ secrets.XXXXXXXX }}"
      STATUS=$(jq -r ".workflow_runs | sort_by( .created_at ) | .[-1] | .status")
      echo "Check suite state:${STATUS}"
</code></pre>

**Comment by gregjwoffindin** [comment]
<p>Thanks, your solution worked and I am receiving the run results.</p>
